tests/cli/commands/lockfile/test_lockfile.py

```diff
@@ -0,0 +1,283 @@
+    assert [(py37, DUAL_PLATFORM_NATIVE_WHEEL.locked_resolves[expected_index])] == list(
+        DUAL_PLATFORM_NATIVE_WHEEL.select([py37, py38])
+    )
+
+
+LOCK_STYLE_SOURCES = lockfile.loads(
+    dedent(
+        """\
+        {
+          "allow_builds": true,
+          "allow_prereleases": false,
+          "allow_wheels": true,
+          "constraints": [],
+          "locked_resolves": [
+            {
+              "locked_requirements": [
+                {
+                  "artifacts": [
+                    {
+                      "algorithm": "sha256",
+                      "hash": "20129f25683fab2099d954379fecd36c13ccc0cc0159eaf59afee53a23d749f1",
+                      "url": "https://files.pythonhosted.org/packages/7c/39/fcd0a978eb327ce8d170ee763264cee1a3a43b0e5f962312d4a37567523d/p537-1.0.4-cp37-cp37m-manylinux1_x86_64.whl"
+                    },
+                    {
+                      "algorithm": "sha256",
+                      "hash": "b1818f434c559706039fa6ca9812f120fc6421b977d5862fb7b411ebaffc074f",
+                      "url": "https://files.pythonhosted.org/packages/56/05/6f01bef57523f6ab7ba5b8fa9831a2204c7ef49dfc194c0d689863f3ae1c/p537-1.0.4.tar.gz"
+                    }
+                  ],
+                  "project_name": "p537",
+                  "requirement": "p537==1.0.4",
+                  "version": "1.0.4",
+                  "via": []
+                }
+              ],
+              "platform_tag": [
+                "cp37",
+                "cp37m",
+                "manylinux_2_33_x86_64"
+              ]
+            }
+          ],
+          "pex_version": "2.1.50",
+          "requirements": [
+            "p537==1.0.4"
+          ],
+          "resolver_version": "pip-legacy-resolver",
+          "transitive": true
+        }
+        """
+    )
+)
+
+
+def test_lockfile_style_sources(
+    py27,  # type: DistributionTarget
+    py37,  # type: DistributionTarget
+    tmpdir,  # type: Any
+):
+    # type: (...) -> None
+
+    selected = {
+        target: locked_resolve for target, locked_resolve in LOCK_STYLE_SOURCES.select([py27, py37])
+    }  # type: Mapping[DistributionTarget, LockedResolve]
+    assert {
+        py27: LOCK_STYLE_SOURCES.locked_resolves[0],
+        py37: LOCK_STYLE_SOURCES.locked_resolves[0],
+    } == selected
+
+    def use_lock(target):
+        # type: (DistributionTarget) -> IntegResults
+        locked_requirements_file = os.path.join(
+            str(tmpdir), "requirements.{}.lock".format(target.id)
+        )
+        with open(locked_requirements_file, "w") as fp:
+            selected[target].emit_requirements(fp)
+        return run_pex_command(
+            args=["-r", locked_requirements_file, "--", "-c", "import p537"],
+            python=target.get_interpreter().binary,
+        )
+
+    use_lock(py37).assert_success()
+
+    # N.B.: We created a lock above that falsely advertises there is a solution for Python 2.7.
```

Comment:
N.B.: Poetry has the same problem. It's inherent to doing anything but creating a strict lock.